packages/pluggableWidgets/carousel-web/src/Carousel.tsx

```diff
@@ -0,0 +1,39 @@
+import { createElement, ReactNode, useEffect, useState, useCallback, ReactElement } from "react";
+import { executeAction } from "@mendix/piw-utils-internal";
+import { CarouselContainerProps } from "../typings/CarouselProps";
+import { Carousel as CarouselComponent } from "./components/Carousel";
+import loadingCircleSvg from "./ui/loading-circle.svg";
+import "./ui/Carousel.scss";
+
+export function Carousel(props: CarouselContainerProps): ReactNode {
+    const { dataSource, content, showPagination, loop, tabIndex, navigation, animation, delay, autoplay } = props;
+    const [loading, setLoading] = useState(true);
+    const onClick = useCallback(() => executeAction(props.onClickAction), [props.onClickAction]);
+    useEffect(() => {
+        if (dataSource?.status === "available") {
```

Comment:
```suggestion
        if (dataSource?.status === ValueStatus.Available) {
```
You can import `ValueStatus` from `mendix`